flink/interpreter/src/main/java/org/apache/zeppelin/flink/JobManager.java

```diff
@@ -218,15 +222,39 @@ public void run() {
       }
     }
 
-      public void cancel () {
-        this.running.set(false);
-        synchronized (running) {
-          running.notify();
-        }
+    public void cancel() {
+      this.running.set(false);
+      synchronized (running) {
+        running.notify();
       }
+    }
 
-      public int getProgress () {
-        return progress;
-      }
+    public int getProgress() {
+      return progress;
     }
   }
+
+  static String toRichTimeDuration(long duration) {
+    long days = TimeUnit.SECONDS.toDays(duration);
```

Comment:
Maybe just use [Duration](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html) data type, and then use corresponding functions, like, `toDays`, `toHours`, ...?

Reply:
It seems using Duration doesn't' save work compared current approach, I still have to get days and minus it from duration, and the same for hours, minutes, seconds. So I didn't change it.

Reply:
Is `String.format("%d:%02d:%02d", s / 3600, (s % 3600) / 60, (s % 60));` to simple?